lms/djangoapps/commerce/views.py

```diff
@@ -0,0 +1,179 @@
+""" Commerce views. """
+
+import logging
+from simplejson import JSONDecodeError
+
+from django.conf import settings
+from django.utils.translation import ugettext as _
+import jwt
+from opaque_keys.edx.keys import CourseKey
+import requests
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.status import HTTP_406_NOT_ACCEPTABLE, HTTP_503_SERVICE_UNAVAILABLE, HTTP_202_ACCEPTED, HTTP_200_OK
+
+from rest_framework.views import APIView
+
+from commerce.constants import OrderStatus
+from course_modes.models import CourseMode
+from courseware import courses
+from student.models import CourseEnrollment
+from util.json_request import JsonResponse
+
+
+log = logging.getLogger(__name__)
+
+
+class DetailResponse(JsonResponse):
+    """ JSON response that simply contains a detail field. """
+
+    def __init__(self, message, status=HTTP_200_OK):
+        data = {'detail': message}
+        super(DetailResponse, self).__init__(object=data, status=status)
+
+
+class ApiErrorResponse(DetailResponse):
+    """ Response returned when calls to the E-Commerce API fail or the returned data is invalid. """
+
+    def __init__(self):
+        message = _('Call to E-Commerce API failed. Order creation failed.')
+        super(ApiErrorResponse, self).__init__(message=message, status=HTTP_503_SERVICE_UNAVAILABLE)
+
+
+class PurchaseView(APIView):
+    """ Purchases course seats and enrolls users. """
+
+    authentication_classes = (SessionAuthentication,)
+    permission_classes = (IsAuthenticated,)
+
+    def _is_data_valid(self, request):
+        """
+        Validates the data posted to the view.
+
+        Arguments
+            request -- HTTP request
+
+        Returns
+            Tuple (data_is_valid, course_key, error_msg)
+        """
+        data = request.DATA or {}
+        course_id = data.get('course_id', None)
+
+        if not course_id:
+            # Translators: Do not translate the term course_id.
+            return False, None, _('course_id field is missing.')
```

Comment:
It would be safer to interpolate the string `course_id` into the translated string. That would prevent translators from accidentally changing it, but would allow them to move it around as necessary.